src/mame/drivers/rastersp.cpp

```diff
@@ -136,30 +157,78 @@ class rastersp_state : public driver_device
 	TIMER_CALLBACK_MEMBER(tms_tx_timer);
 	DECLARE_WRITE_LINE_MEMBER(vblank_irq);
 
-	std::unique_ptr<uint8_t[]>   m_nvram8;
-	uint8_t   m_irq_status;
-	uint32_t  m_dpyaddr;
-	std::unique_ptr<uint16_t[]> m_paletteram;
-	uint32_t  m_speedup_count;
-	uint32_t  m_tms_io_regs[0x80];
-	bitmap_ind16 m_update_bitmap;
-
 	uint32_t screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
-	void update_irq(uint32_t which, uint32_t state);
 	void upload_palette(uint32_t word1, uint32_t word2);
 	IRQ_CALLBACK_MEMBER(irq_callback);
 	nscsi_connector &add_rastersp_scsi_slot(machine_config &config, const char *tag, const char *default_slot);
 	static void ncr53c700_config(device_t *device);
 	void cpu_map(address_map &map);
 	void dsp_map(address_map &map);
 	void io_map(address_map &map);
+	uint8_t interrupt_status_r(offs_t offset);
+	
+	std::unique_ptr<uint8_t[]>   m_nvram8;
+	uint8_t   m_irq_status;
+	uint32_t  m_dlba;
+	std::unique_ptr<uint16_t[]> m_paletteram;
+	uint8_t m_palette_number;
+	uint32_t  m_speedup_count;
+	uint32_t  m_tms_io_regs[0x80];
+	bitmap_ind16 m_update_bitmap;
+	
+	uint8_t m_port2_data;
+	uint32_t m_last_watchdog;
+	int m_left_volume;
+	int m_right_volume;
+	uint8_t m_interrupt_mask;
+	uint8_t m_dsp_ctrl_data;
+};
+
+class fbcrazy_state : public rastersp_state
+{
+public:
+	fbcrazy_state(const machine_config &mconfig, device_type type, const char *tag)
+		: rastersp_state(mconfig, type, tag)
+		, m_edc_tb_timer(nullptr)
+	{
+	}
+
+	void fbcrazy(machine_config &config);
 
-	// driver_device overrides
+	DECLARE_READ_LINE_MEMBER(meter_pulse_r);
+
+protected:
 	virtual void machine_reset() override;
 	virtual void machine_start() override;
-	virtual void video_start() override;
-};
 
+private:
+
+	emu_timer *m_edc_tb_timer;
+	void aux_port0_w(offs_t offset, uint8_t data);
+	void aux_port1_w(offs_t offset, uint8_t data);
+	void aux_port2_w(offs_t offset, uint8_t data);
+	void aux_port3_w(offs_t offset, uint8_t data);
+	void aux_port4_w(offs_t offset, uint8_t data);
+	uint8_t m_aux_port3_data;
```

Comment:
Please group data members.

Reply:
To be sure, you mean move the m_aux_port3_data and m_edc_tb_timer so they're together with m_uart_regs etc ?
So not intermixed with function definitions ?
Have done that locally.